src/App.tsx

```diff
@@ -1,14 +1,33 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState(true);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
```

Comment:
you don't need this state.
It's better to move local state of filters to app component and just create a var to store filtered todos. and your filter will just set the proper values. All todo manipulations will remain in App component since it is single source of truth